hedgehog/src/Hedgehog/Internal/Runner.hs

```diff
@@ -111,17 +112,18 @@ takeSmallest ::
   -> ShrinkLimit
   -> ShrinkRetries
   -> (Progress -> m ())
-  -> Node m (Maybe (Either Failure (), [Log]))
+  -> Node m (Maybe (Either Failure (), (PropResult, [Log])))
   -> m Result
 takeSmallest size seed shrinks slimit retries updateUI = \case
   Node Nothing _ ->
     pure GaveUp
 
-  Node (Just (x, w)) xs ->
+  Node (Just (x, (_, w))) xs ->
     case x of
       Left (Failure loc err mdiff) -> do
         let
           failure =
+            -- TODO: maybe add classifier to failure?
```

Comment:
We use `FIXME` throughout the code base to mark something to be done. See similar nitpick above (shouldn't be a blocker, of course).